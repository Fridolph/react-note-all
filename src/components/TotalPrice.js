import React from 'react'
import PropTypes from 'prop-types'

function TotalPrice({income, outcome}) {
  return (
    <div className="container-fluid">
      <div className="row">
        <h4 className="col" style={{textAlign: 'center', margin: '30px 0'}}>
          收入：{income}
        </h4>
        <h4 className="col" style={{textAlign: 'center', margin: '30px 0'}}>
          支出：{outcome}
        </h4>
      </div>
    </div>
  )
}

TotalPrice.propTypes = {
  income: PropTypes.number.isRequired,
  outcome: PropTypes.number.isRequired,
}

export default TotalPrice
