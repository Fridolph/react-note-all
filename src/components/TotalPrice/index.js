import React from 'react'
import PropTypes from 'prop-types'

function TotalPrice({income, outcome}) {
  return (
    <div className="container-fluid">
      <div className="row">
        <h4 className="col income" style={{textAlign: 'center', margin: '30px 0'}}>
          收入：<span>{income}</span>
        </h4>
        <h4 className="col outcome" style={{textAlign: 'center', margin: '30px 0'}}>
          支出：<span>{outcome}</span>
        </h4>
      </div>
    </div>
  )
}

TotalPrice.propTypes = {
  income: PropTypes.number.isRequired,
  outcome: PropTypes.number.isRequired,
}

export default TotalPrice
