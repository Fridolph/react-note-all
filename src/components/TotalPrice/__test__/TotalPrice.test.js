import React from 'react'
import { shallow } from 'enzyme'
import TotalPrice from '../'

const props = {
  income: 1000,
  outcome: 2000
}

let wrapper

describe('@src/components/TotalPrice', () => {
  beforeEach(() => {
    wrapper = shallow(<TotalPrice {...props} />)
  })

  it('snapshot能匹配', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('component should render correct income & outcome number', () => {
    const wrapper = shallow(<TotalPrice {...props} />)
    expect(wrapper.find('.income span').text() * 1).toEqual(1000)
    expect(wrapper.find('.outcome span').text() * 1).toEqual(2000)
  })
})
