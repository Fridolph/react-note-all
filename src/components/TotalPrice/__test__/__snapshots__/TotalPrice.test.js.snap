// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@src/components/TotalPrice snapshot能匹配 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalPrice
    income={1000}
    outcome={2000}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <h4
          className="col income"
          style={
            Object {
              "margin": "30px 0",
              "textAlign": "center",
            }
          }
        >
          收入：
          <span>
            1000
          </span>
        </h4>
        <h4
          className="col outcome"
          style={
            Object {
              "margin": "30px 0",
              "textAlign": "center",
            }
          }
        >
          支出：
          <span>
            2000
          </span>
        </h4>
      </div>,
      "className": "container-fluid",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="col income"
            style={
              Object {
                "margin": "30px 0",
                "textAlign": "center",
              }
            }
          >
            收入：
            <span>
              1000
            </span>
          </h4>,
          <h4
            className="col outcome"
            style={
              Object {
                "margin": "30px 0",
                "textAlign": "center",
              }
            }
          >
            支出：
            <span>
              2000
            </span>
          </h4>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "收入：",
              <span>
                1000
              </span>,
            ],
            "className": "col income",
            "style": Object {
              "margin": "30px 0",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "收入：",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1000,
              },
              "ref": null,
              "rendered": 1000,
              "type": "span",
            },
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "支出：",
              <span>
                2000
              </span>,
            ],
            "className": "col outcome",
            "style": Object {
              "margin": "30px 0",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "支出：",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2000,
              },
              "ref": null,
              "rendered": 2000,
              "type": "span",
            },
          ],
          "type": "h4",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <h4
            className="col income"
            style={
              Object {
                "margin": "30px 0",
                "textAlign": "center",
              }
            }
          >
            收入：
            <span>
              1000
            </span>
          </h4>
          <h4
            className="col outcome"
            style={
              Object {
                "margin": "30px 0",
                "textAlign": "center",
              }
            }
          >
            支出：
            <span>
              2000
            </span>
          </h4>
        </div>,
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="col income"
              style={
                Object {
                  "margin": "30px 0",
                  "textAlign": "center",
                }
              }
            >
              收入：
              <span>
                1000
              </span>
            </h4>,
            <h4
              className="col outcome"
              style={
                Object {
                  "margin": "30px 0",
                  "textAlign": "center",
                }
              }
            >
              支出：
              <span>
                2000
              </span>
            </h4>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "收入：",
                <span>
                  1000
                </span>,
              ],
              "className": "col income",
              "style": Object {
                "margin": "30px 0",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              "收入：",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1000,
                },
                "ref": null,
                "rendered": 1000,
                "type": "span",
              },
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "支出：",
                <span>
                  2000
                </span>,
              ],
              "className": "col outcome",
              "style": Object {
                "margin": "30px 0",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              "支出：",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 2000,
                },
                "ref": null,
                "rendered": 2000,
                "type": "span",
              },
            ],
            "type": "h4",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
